src/app/products/manage-tax-components/create-tax-component/create-tax-component.component.ts

```diff
@@ -0,0 +1,129 @@
+  styleUrls: ['./create-tax-component.component.scss']
+})
+export class CreateTaxComponentComponent implements OnInit {
+
+  /** Minimum start date allowed. */
+  minDate = new Date();
+  /** Maximum start date allowed. */
+  maxDate = new Date(2025, 6, 22);
+  /** Tax Component form. */
+  taxComponentForm: FormGroup;
+  /** Tax Component template data. */
+  taxComponentTemplateData: any;
+  /** Credit Account Type data. */
+  creditAccountTypeData: any;
+  /** Credit Account data. */
+  creditAccountData: any;
+  /** Debit Account Type data. */
+  debitAccountTypeData: any;
+  /** Debit Account data. */
+  debitAccountData: any;
+
+  /**
+   * Retrieves the tax Component template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Products Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe(( data: { taxComponentTemplate: any }) => {
+      this.taxComponentTemplateData = data.taxComponentTemplate;
+    });
+  }
+
+  /**
+   * Creates the tax Component form
+   */
+  ngOnInit() {
+    this.createTaxComponentForm();
+    this.setConditionalControls();
+  }
+
+  /**
+   * Creates the tax Component form
+   */
+  createTaxComponentForm() {
+    this.creditAccountTypeData = this.debitAccountTypeData = this.taxComponentTemplateData.glAccountTypeOptions;
+    this.taxComponentForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'percentage': ['', [Validators.required, Validators.min(1), Validators.max(100)]],
+      'creditAccountType': [''],
+      'debitAccountType': [''],
+      'startDate': ['', Validators.required],
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the tax Component form
+   */
+  setConditionalControls() {
+    this.taxComponentForm.get('debitAccountType').valueChanges.subscribe(debitAccountTypeId => {
+      this.debitAccountData = this.getAccountsData(debitAccountTypeId);
+      this.taxComponentForm.addControl('debitAcountId', new FormControl('', Validators.required));
+    });
+    this.taxComponentForm.get('creditAccountType').valueChanges.subscribe(creditAccountTypeId => {
+      this.creditAccountData = this.getAccountsData(creditAccountTypeId);
+      this.taxComponentForm.addControl('creditAcountId', new FormControl('', Validators.required));
+    });
+  }
+
+  /**
+   * @param {number} accountTypeId Account type ID of account type.
+   * @returns {any} Accounts data
+   */
+  getAccountsData(accountTypeId: number) {
+    const glAccountOptionsData = this.taxComponentTemplateData.glAccountOptions;
```

Comment:
Can't this be defined at the top of the file?

Reply:
This variable only has one use case in that function, it'll be useless to make it a global variable.

Reply:
No. We can remove this variable though as its extra and use `taxComponentTemplateData` directly.

Reply:
okay